lib/waterline/query/aggregate.js

```diff
@@ -51,8 +53,49 @@ module.exports = {
       return value !== undefined;
     });
     
-    // Create will take care of cloning values so original isn't mutated
-    async.map(filteredValues, self.create.bind(self), cb);
+    // process each item, their associations and beforeCreate callbacks
```

Comment:
I think we don't need to check for it again here, as the check is already in place inside the adapter handler: https://github.com/masumsoft/waterline/blob/createeach-fix/lib/waterline/adapter/aggregateQueries.js#L25

The call to self.adapter.createEach() method in my implementation actually calls the underlined createEach() method inside adapter/aggregateQueries.js, which already handles if the adapter implements createEach or not.